src/ChapterUtils.cs

```diff
@@ -24,6 +26,14 @@ internal class ChapterUtils
         LevelID.GothicSandbox,
         LevelID.DogSandbox
     ];
+    // Cache filenames with their full file paths. The functionality of this cache assumes that there are
+    // no dynamic file changes that occur while the game is running. So the cache will be cleared when the game is restarted.
```

Comment:
I'm not sure this assumption is reasonable. In particular it might make developing chapters more cumbersome, since you'd need to restart the game every time you want to add a new level.